src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ModelBindingHelper.cs

```diff
@@ -744,35 +743,6 @@ internal static void ValidateBindingContext(ModelBindingContext bindingContext)
             }
         }
 
-        internal static void ValidateBindingContext(
-            ModelBindingContext bindingContext,
-            Type requiredType,
-            bool allowNullModel)
-        {
-            ValidateBindingContext(bindingContext);
-
-            if (bindingContext.ModelType != requiredType)
-            {
-                var message = Resources.FormatModelBinderUtil_ModelTypeIsWrong(bindingContext.ModelType, requiredType);
-                throw new ArgumentException(message, nameof(bindingContext));
-            }
-
-            if (!allowNullModel && bindingContext.Model == null)
-            {
-                var message = Resources.FormatModelBinderUtil_ModelCannotBeNull(requiredType);
-                throw new ArgumentException(message, nameof(bindingContext));
-            }
-
-            if (bindingContext.Model != null &&
-                !bindingContext.ModelType.GetTypeInfo().IsAssignableFrom(requiredType.GetTypeInfo()))
```

Comment:
`!bindingContext.ModelType.GetTypeInfo().IsAssignableFrom(requiredType.GetTypeInfo())` is always `false` because `bindingContext.ModelType == requiredType` to reach here. (Yeah, this method only dealt correctly with `sealed` types anyhow.)